src/librustc_resolve/diagnostics.rs

```diff
@@ -948,6 +948,50 @@ use something_which_doesnt_exist;
 Please verify you didn't misspell the import's name.
 "##,
 
+E0434: r##"
+This error indicates that a variable usage inside an inner function is invalid
+because the variable comes from a dynamic environment. Inner functions do not
+have access to their containing environment.
+Example of erroneous code:
+
+```compile_fail
+fn foo() {
+    let y = 5;
+    fn bar() -> u32 {
+        y // error: can't capture dynamic environment in a fn item; use the
+          //        || { ... } closure form instead.
+    }
+}
+```
+
+Functions do not capture local variables. To fix this error, please replace the
+function with a closure, or the variable with a constant or static item.
+Example:
+
+```
+fn foo() {
+    let y = 5;
+    let bar = || {
+        y
+    };
+}
+```
+In some cases, it may be possible to change the variable to a constant or
+static item.
```

Comment:
I agree that it should be "a static item".  It's worth mentioning that the form that I had used was also used in other error messages.
A few examples - [E0020](https://doc.rust-lang.org/error-index.html#E0020), [E0063](https://doc.rust-lang.org/error-index.html#E0063), [E0071](https://doc.rust-lang.org/error-index.html#E0071), [E0106](https://doc.rust-lang.org/error-index.html#E0106)